main_dialog.py

```diff
@@ -0,0 +1,308 @@
+            print(self.list_widget_1.item(x).showExtension())
+
+    def dragEnterEvent(self, e):
+        e.accept()
+
+    def dropEvent(self, e):
+        self.items_changed()
+
+# << ADD paths on list_widget_1 from listPaths used pop()  >> #
+    def add_to_list_widget(self, listPaths):
+        for path in listPaths:
+            fileName = str(path).split("/")
+            item = MyListWidgetItem(path)
+            item.setText(fileName[-1])
+            item.setPath(str(path))
+            self.list_widget_1.addItem(item)
+        # for debugging
+        self.print_extensions()
+        self.check_extensions()
+
+# << Load files on >> #
+
+    def load_files(self):
+        listPaths = []
+        files, _ = QFileDialog.getOpenFileNames(
+                self, "Wybierz pliki", '',
+                "wszystkie (*);; " + "commonmark (*.commonmark);;" +
+                "docbook (*.docbook);;" + "docx (*.docx);;" +
+                "epub (*.epub);;" + "haddock (*.haddock);;" +
+                "html (*.html);;" + "json (*.json);;" + "latex (*.latex);;" +
+                "markdown (*.markdown *.md);;" +
+                "markdown_github (*.markdown_github);;" +
+                "markdown_mmd(*.markdown_mmd);;" +
+                "markdown_phpextra (*.markdown_phpextra);;" +
+                "markdown_strict (*.markdown_strict);;" +
+                "mediawiki (*.mediawiki);;" + "native (*.native);;" +
+                "odt (*.odt);;" + "opml (*.opml);;" + "org (*.org);;" +
+                "rst (*.rst);;" + "t2t (*.t2t);;" + "textile (*.textile);;" +
+                "twiki (*.twiki)")
+
+        for file in files:
+            listPaths.append(file)
+        print(listPaths)  # for debugging
+        self.add_to_list_widget(listPaths)
+        self.items_changed()
+# << END of: Load files on >> #
+
+    def return_boxIsChecked(self):
+        self.isChecked = self.check_box_1.checkState()
+        return self.isChecked
+
+    def return_files(self):
+        self.returnedFiles = []
+        for x in range(self.list_widget_1.count()):
+            self.returnedFiles.append(self.list_widget_1.item(x).showPath())
+        return self.returnedFiles
+# << >>
+
+    # << Select recipe - handling >> #
+    def select_recipe(self):
+        recipeDialog = None
+        recipeDialog = RecipeDialog(recipeDialog, self.selectedRecipe)
+        recipeDialog.exec_()
+        self.selectedRecipe = recipeDialog.retRecipe()
+        self.items_changed()
+        print(self.selectedRecipe)
+        # for debugging
+
+    # << END of: Select recipe - handling >>
+    # with ZipFile('spam.zip') as myzip:
+
+    def conf_variables(self):
+        bookName = str(self.line_edit_1.text())
+        variablesDialog = VariablesDialog(self.selectedRecipe,
+                                          self.return_files(),
+                                          self.return_boxIsChecked(),
+                                          bookName)
```

Comment:
To wywołanie metody jako argument brzydko wygląda. Utworzyłbym zmienne przed tym wywołaniem.